db/db_impl/db_impl.cc

```diff
@@ -1356,13 +1356,21 @@ Status DBImpl::ApplyReplicationLogRecord(ReplicationLogRecord record,
 
         auto current_update_sequence = versions_->GetManifestUpdateSequence();
         uint64_t latest_applied_update_sequence = 0;
+        auto replication_epoch =
+            immutable_db_options_.replication_log_listener
+                ->EpochOfReplicationSequence(replication_sequence);
```

Comment:
At this point I wonder if we should just move the RS serialization into RocksDB-Cloud.

Reply:
hmm this seems to be slightly more complex than I expected. I'll make a separate PR for this.

Reply:
Yeah that's okay